src/sessions.ts

```diff
@@ -299,11 +299,8 @@ export class ClientSession
         if (serverSession != null) {
           // release the server session back to the pool
           this.sessionPool.release(serverSession);
-          // Make sure a new serverSession never makes it onto this ClientSession
-          Object.defineProperty(this, kServerSession, {
-            value: ServerSession.clone(serverSession),
-            writable: false
```

Comment:
This is an over abundance of caution for something we can either enforce with a flag or tolerably let be, this symbol property is internal an unlikely to be written to after this point.

Reply:
Can we make the property private and use a `ts-expect-error` here with a comment explaining why we're assigning the readonly property?

Reply:
I can make it private (it is already our implicit form of private b/c its symbol and internal)

But I don't understand the readonly suggestion since it is a property we have to write to. What's the benefit of marking it readonly only to ignore that?

Reply:
you implemented this to make it not writable by defining it as `writable: false`.  But you're removing that restriction.

An alternative approach is to make it readonly so that future changes can't write to this value without TS errors.

Reply:
It was not writable after this point in time, which is different than a property being readonly upon creation. 

If I make it readonly on the class I also have to ignore the error that shows up where we actually do write to the property as intended. I think that is undesirable since expect-error ignores _all_ ts checking not only the thing we're thinking about now. Also, it is the wrong intention, it is intentionally writable because it's lazily calculated to avoid TooManyLogicalSessions errors.